File: src/main/java/io/github/jakejmattson/touchcontrol/display/ImageFrame.java
package io.github.jakejmattson.touchcontrol.display;

import javax.swing.JFrame;
import java.awt.*;
import java.awt.event.*;
import java.awt.image.BufferedImage;

/**
 * Frame - GUI container for components (holds ImagePanel).
 *
 * @author JakeJMattson
 */
public class ImageFrame
{
	private final JFrame frame;
	private final ImagePanel imagePanel;
	private boolean isOpen;

	public ImageFrame(String title)
	{
		imagePanel = new ImagePanel();
		isOpen = true;

		frame = new JFrame(title);
		frame.addWindowListener(createWindowListener());
		frame.add(imagePanel);
		frame.setVisible(true);
	}

	public boolean isOpen()
	{
		return isOpen;
	}

	/**
	 * Create a listener to monitor the frame closing event.
	 *
	 * @return WindowListener
	 */
	private WindowListener createWindowListener()
	{
		return new WindowAdapter()
		{
			@Override
			public void windowClosing(WindowEvent windowClosed)
			{
				isOpen = false;
			}
		};
	}

	/**
	 * Display an image in the frame.
	 *
	 * @param image
	 * 		Image to be shown
	 */
	public void showImage(BufferedImage image)
	{
		imagePanel.setImage(image);
		frame.repaint();
		frame.pack();
	}

	public void setLocation(Point location)
	{
		frame.setLocation(location);
	}
}

File: src/main/kotlin/me/jakejmattson/touchcontrol/display/ImageFrame.kt
package me.jakejmattson.touchcontrol.display

import java.awt.Point
import java.awt.event.*
import java.awt.image.BufferedImage
import javax.swing.*

/**
 * Frame - GUI container for components (holds ImagePanel).
 *
 * @author JakeJMattson
 */
class ImageFrame(title: String) {
    private val leftPanel = ImagePanel()
    private val rightPanel = ImagePanel()

    private val frame: JFrame = JFrame(title).apply {
        addWindowListener(createWindowListener())

        add(JPanel().apply {
            add(leftPanel)
            add(rightPanel)
        })

        isVisible = true
    }

    var isOpen: Boolean = true
        private set

    private fun createWindowListener(): WindowListener {
        return object : WindowAdapter() {
            override fun windowClosing(windowClosed: WindowEvent?) {
                isOpen = false
            }
        }
    }

    fun showImages(leftImage: BufferedImage, rightImage: BufferedImage) {
        leftPanel.image = leftImage
        rightPanel.image = rightImage
        frame.repaint()
        frame.pack()
    }

    fun setLocation(location: Point) {
        frame.location = location
    }
}